app/screens/more_channels/more_channels.js

```diff
@@ -152,6 +161,11 @@ export default class MoreChannels extends PureComponent {
                     currentTeamId,
                     this.page + 1,
                     General.CHANNELS_CHUNK_SIZE
+                );
+                actions.getArchivedChannels(
```

Comment:
this call should only be executed if `canShowArchivedChannels` is set to true.

Also I was asking why it had to be sequential before cause I think you can do this in a new action and call both of them when needed instead of calling them separately here.

The problem this is now presenting is that you are calling `this.loadChannels` when fetching the archived channels finishes but it may be that `actions.getChannels` has not yet fetched anything yet.

If you want we can talk more about this before moving forward

Reply:
No, that makes sense - you're right - I'll fix it up!

Reply:
So I've created a new action called `loadPublicAndArchivedChannels` that first fires the `getChannels` action, and then if `canShowArchivedChannels` is true, it also fires the `getArchivedChannels` action, and then finally returns that promise. It should now call `loadChannels` in the `then()` after the public channels are loaded, or after the public channels & the archived channels are loaded. Is this better? Let me know!